rero_ils/modules/notifications/api.py

```diff
@@ -65,195 +61,148 @@ class Meta:
         default_filter = None
 
 
-class Notification(IlsRecord):
-    """Notifications class."""
+class Notification(IlsRecord, ABC):
+    """Notifications class.
+
+    A Notification is an abstract class representing a message to be sent to a
+    recipient. All the notification workflow depends of the notification type.
+    The recipient, the dispatcher method, notification aggregation... all
+    these settings or behaviors must be specified into a concrete
+    ``Notification`` subclass.
+
+    But all notifications (whatever the notification type) are stored in the
+    same resource and share the same JSON schema. So, this parent class defines
+    every Invenio resource configuration : minters, fetchers, API methods...
+
+    When a ``Notification`` is created or loaded, this is the
+    ``NotificationSubclassExtension`` extension which determine the best
+    possible concrete subclass (see this extension for more information).
+
+      # >>> notif = Notification.get_record_by_pid('n1')
+      # >>> print(type(notif))
+      # >>>   <class 'ConcreteNotificationSubclass'>
+
+    """
 
     minter = notification_id_minter
     fetcher = notification_id_fetcher
     provider = NotificationProvider
     model_cls = NotificationMetadata
 
+    _extensions = [
+        NotificationSubclassExtension()
+    ]
+
+    # INVENIO API METHODS =====================================================
+    #   Override some invenio ``RecordBase`` method
     @classmethod
-    def create(cls, data, id_=None, delete_pid=False,
-               dbcommit=False, reindex=False, **kwargs):
+    def create(cls, data, id_=None, delete_pid=False, dbcommit=False,
+               reindex=False, **kwargs):
         """Create notification record."""
-        data.setdefault('status', 'created')
-        record = super().create(
-            data, id_, delete_pid, dbcommit, reindex, **kwargs)
+        data.setdefault('status', NotificationStatus.CREATED)
```

Comment:
It could... but it's maybe a little bit overkill for just 1 line of code.
I need to keep the overrided `create` method because, `PatronTransaction` creation cannot be moved into the `post_create` hook (I need dbcommit, reindex and delete_pid params)